### DISCUSSION TITLE
Gemm Fused Row Reduction question.

### DISCUSSION THREAD

**Discussion by MARD1NO** [NVIDIA/cutlass#712] [Q&A] [answered]
Recently I read `35_gemm_softmax` example, and I wonder know if my understanding is right. 

Since softmax is a row reduction, we compute each block tile, and maintain the tile row_max_val. After we finish a row result, we reduce each block tile's row_max_val(is finished by ApplyFinalReductionKernel), then we launch `SoftmaxApplyKernel` to calculate final result.

**Comment by hwu36** [comment, marked as answer]
You are correct.
